### PR TITLE
Fixup broken links in docs

### PR DESCRIPTION
**Pull request by trey-stafford**
Fixes a couple of broken links that I came across while browsing the `earthaccess` documentation. Resolves https://github.com/nsidc/earthaccess/issues/778

- [x] Please review our
      [contributing documentation](https://earthaccess.readthedocs.io/en/latest/contributing/)
      before getting started.
- [x] Ensure an issue exists representing the problem being solved in this PR.
- [x] Populate a descriptive title. For example, instead of "Updated README.md", use a
      title such as "Add testing details to the contributor section of the README".
- [x] Populate the body of the pull request with:
    - A clear description of the change you are proposing.
    - Links to any issues resolved by this PR with text in the PR description, for
      example `closes #1`. See
      [GitHub docs - Linking a pull request to an issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
- [ ] Update `CHANGELOG.md` with details about your change in a section titled
      `## Unreleased`. If such a section does not exist, please create one. Follow
      [Common Changelog](https://common-changelog.org/) for your additions.
- [x] Update the documentation and/or the `README.md` with details of changes to the
      earthaccess interface, if any. Consider new environment variables, function names,
      decorators, etc.

Click the "Ready for review" button at the bottom of the "Conversation" tab in GitHub
once these requirements are fulfilled. Don't worry if you see any test failures in
GitHub at this point!

</details>

<details><summary>Pull Request (PR) merge checklist - click to expand</summary>

Please do your best to complete these requirements! If you need help with any of these
requirements, you can ping the `@nsidc/earthaccess-support` team in a comment and we
will help you out!

- [x] Add unit tests for any new features.
- [x] Apply formatting and linting autofixes. You can add a GitHub comment in this Pull
      Request containing "pre-commit.ci autofix" to automate this.
- [x] Ensure all automated PR checks (seen at the bottom of the "conversation" tab) pass.
- [ ] Get at least one approving review.

</details>

----
📚 Documentation preview 📚: https://earthaccess--777.org.readthedocs.build/en/777/

### CODE REVIEW

File: docs/user_guide/authenticate.md [hunk 1]
```diff
@@ -6,5 +6,6 @@ Introduces the `earthaccess.login` method for managing Earthdata Login and cloud
 
     We are reorganizing and updating the documentation, so not all pages are complete.  If you are looking for information about authenticating using earthaccess see the
     How-Tos and Tutorials in links below.
-    [How-To Authenticate with earthaccess](/howto/authenticate.md)
-    [Getting Started Tutorial](/tutorials/getting-started.ipynb)
+
+    * [Quick start](../../quick-start/)
```
Comment (trey-stafford) [inline docs/user_guide/authenticate.md:10]: The "getting started" tutorial doesn't appear to exist anymore, so I linked to the 'quick-start'.

**Comment by trey-stafford** [comment]
I'm noticing now that the other "user guide" pages also have this problem.

I also just noticed another broken link on this page: https://earthaccess.readthedocs.io/en/latest/howto/search-granules/ ("Introducing NASA earthaccess" tutorial link).

**Comment by trey-stafford** [comment]
> I'm noticing now that the other "user guide" pages also have this problem.

I updated those pages w/ new commits to this PR.

> I also just noticed another broken link on this page: https://earthaccess.readthedocs.io/en/latest/howto/search-granules/ ("Introducing NASA earthaccess" tutorial link).

I'm not sure what this is supposed to link to. The URL is https://nsidc.github.io/earthaccess/tutorials/demo/#querying-for-datasets, but I do not see any "demo" directory under tutorials.

**Review by mfisher87 — APPROVED** [review]
LGTM!  Thank you, Trey :)

**Comment by mfisher87** [comment]
I wonder why our build process didn't catch these broken links. It caught others! :thinking: 

Anyway, congrats on opening lucky PR #777 :partying_face:

**Pull request merged**